### DISCUSSION TITLE
Unify behaviour of '''hessian''' for univariate and multivariate functions.

### DISCUSSION THREAD

**Discussion by Emilius12** [sympy/sympy#26485] [Ideas]
When working on [this](https://github.com/sympy/sympy/pull/26450) pull request, I encountered an error while attempting to compute the Hessian matrix of a univariate function. Upon investigation, I found that this error was introduced in [this](https://github.com/sympy/sympy/commit/081e40b1e37350d64c50b80cf3949967e818542a#diff-0abe7f67dba49155581cc51d6916abab407a75720a74a79ba9360ebf16a56a07) commit. I questioned whether it was advisable to retain this error. Modifying the behavior of this function seemed straightforward, but lacking in-depth knowledge of the potential ramifications of such a change, I sought the community's input.

**Comment by oscarbenjamin** [comment]
Can you explain what the problem is that is caused by that commit?

**Reply by Emilius12** [reply]
The issue is that the function '''hessian''' cannot work for univariate functions. My current solution for implementing convexity check for multivariate function is to discriminate cases but it would be better to have a unique behaviour

**Comment by oscarbenjamin** [comment]
It seems to work for univariate functions:
```python
In [8]: hessian(f(x), [x])
Out[8]:
⎡ 2       ⎤
⎢d        ⎥
⎢───(f(x))⎥
⎢  2      ⎥
⎣dx       ⎦

In [9]: hessian(f(x, y), [x, y])
Out[9]:
⎡  2               2           ⎤
⎢ ∂               ∂            ⎥
⎢ ───(f(x, y))   ─────(f(x, y))⎥
⎢   2            ∂y ∂x         ⎥
⎢ ∂x                           ⎥
⎢                              ⎥
⎢  2               2           ⎥
⎢ ∂               ∂            ⎥
⎢─────(f(x, y))   ───(f(x, y)) ⎥
⎢∂y ∂x              2          ⎥
⎣                 ∂y           ⎦
```

**Reply by Emilius12** [reply]
Okay, I didn't realize that you had to call` hessian(f, [x])` instead of `hessian(f, x)`. Thanks for pointing that out! I think I'll be able to implement convexity checks for both univariate and multivariate functions now. Sorry for any inconvenience caused.
